vendor/github.com/prometheus/alertmanager/config/notifiers.go

```diff
@@ -796,5 +810,17 @@ type MSTeamsConfig struct {
 func (c *MSTeamsConfig) UnmarshalYAML(unmarshal func(interface{}) error) error {
 	*c = DefaultMSTeamsConfig
 	type plain MSTeamsConfig
-	return unmarshal((*plain)(c))
+	if err := unmarshal((*plain)(c)); err != nil {
+		return err
+	}
+
+	if c.WebhookURL == nil && c.WebhookURLFile == "" {
+		return fmt.Errorf("one of webhook_url or webhook_url_file must be configured")
+	}
+
+	if c.WebhookURL != nil && len(c.WebhookURLFile) > 0 {
```

Comment:
We have validation for _some_ notifiers in `pkg/alertmanager/api.go` that ensures they have not set configuration that reads arbitrary files from disk. This seems like it's adding new configuration that causes files to be read off disk. We need to add checks to the validation in `api.go` unless I'm misunderstanding what this is.